ios-training/WeatherView/WeatherPresenter.swift

```diff
@@ -42,28 +41,28 @@ final class WeatherPresenter: WeatherPresenterInput {
     }
 }
 
-// MARK: WeatherInfoRepositoryDelegate
-
-extension WeatherPresenter: WeatherInfoRepositoryDelegate {
-    func didFetch(result: Result<WeatherInfo, Error>) {
-        DispatchQueue.main.async {
-            switch result {
-            case .success(let weatherInfo):
-                self.view.showWeatherInfo(weatherInfo: weatherInfo)
-            case .failure:
-                self.view.showFetchErrorAlert()
-            }
-            self.view.finishLoading()
-        }
-    }
-}
-
 // MARK: - Private
 
 private extension WeatherPresenter {
     /// 天気を読み込む
     func loadWeather() {
         view.startLoading()
-        weatherInfoRepository.requestFetch(at: "tokyo", date: Date())
+        weatherInfoRepository.fetch(at: "tokyo", date: Date()) { [weak self] result in
+            guard let `self` else { return }
```

Comment:
[must] ``` `self` ```できるのはbugなので使わない
- https://github.com/apple/swift-evolution/blob/main/proposals/0079-upgrade-self-from-weak-to-strong.md#relying-on-a-compiler-bug
```suggestion
            guard let self else { return }
```

Reply:
有益な情報をありがとうございます！
対応しました